src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -114,11 +125,25 @@ private void Initialize(Compilation comp, AnalyzerExecutor analyzerExecutor, Can
             }
         }
 
-        private Task ExecuteSyntaxTreeActions(CancellationToken cancellationToken)
+        public void StartCompleteAnalysis(ImmutableHashSet<SyntaxTree> syntaxTreesToSkip, CancellationToken cancellationToken)
+        {
+            if (_initializeTaskStarted && _primaryTask == null)
+            {
+                _primaryTask = Task.Run(async () =>
+                    {
+                        await _initializeTask.ConfigureAwait(false);
+                        await ProcessCompilationEventsAsync(true, cancellationToken).ConfigureAwait(false);
+                        await ExecuteSyntaxTreeActions(_compilation.SyntaxTrees.Where((tree) => !syntaxTreesToSkip.Contains(tree)), cancellationToken).ConfigureAwait(false);
```

Comment:
I should have added comments about it :) Probably you can add one given you are touching this code..